### DISCUSSION TITLE
How to change a node name to another?

### DISCUSSION THREAD

**Discussion by paring-chan** [k3s-io/k3s#6547] [Q&A]
I have a k3s node that matches with my domain name. And I found the dns lookup is not working properly because of that. So I have to change the name of k3s node. What can I do for this?

**Comment by brandond** [comment]
If it's the only server node in the cluster, just change the hostname, restart the service, and use `kubectl delete node` to clean up the old entry.

If there are other server nodes, then change the hostname, stop the service on the affected node, use `kubectl delete node` on another node to delete it, then start the service again.

If you have agents pointed at that server, update their configuration to point at the correct host, and restart them.
